misc/identity-keys/src/identity-keys.ts

```diff
@@ -96,30 +109,28 @@ export class IdentityKeys implements IIdentityKeys {
 
         const signature = await onSign(cacaoMessage);
 
-        // Storing keys after signature creation to prevent having false statement
-        // Eg, onSign failing / never resolving but having identity keys stored.
-        this.identityKeys.set(accountId, {
-          identityKeyPriv: privKeyHex,
-          identityKeyPub: pubKeyHex,
-          accountId,
-        });
+        if (!signature) {
+          throw new Error("Provided an empty signature");
+        }
 
         const url = `${this.keyserverUrl}/identity`;
 
-        const response = await axios.post(url, {
-          cacao: {
-            ...cacao,
-            s: {
-              ...cacao.s,
-              s: signature,
+        try {
+          await axios.post(url, {
+            cacao: {
+              ...cacao,
+              s: {
+                ...cacao.s,
+                s: signature,
+              },
             },
-          },
-        });
-
-        if (response.status !== 200) {
-          throw new Error(`Failed to register on keyserver ${response.status}`);
+          });
+        } catch (e) {
+          throw new Error(`Failed to register on keyserver`);
```

Comment:
Should log what happened with the keyserver:

```suggestion
          throw new Error(`Failed to register on keyserver: ${e}`);
```